src/linear_regression.cpp

```diff
@@ -84,8 +84,8 @@ Rcpp::List linear_regression(Eigen::Map<Eigen::MatrixXd> X,
     }
 
     Rcpp::List res = Rcpp::List::create(Rcpp::Named("loss") = loss,
-                                        Rcpp::Named("theta") = theta_hat_prev,
-                                        Rcpp::Named("gradient") = theta_adj_prev,
+                                        Rcpp::Named("theta") = theta_hat_prev.col(0),
```

Comment:
Hm I would've assumed this is not necessary because `theta_hat_prev` is already declared as a vector 🤔

Reply:
It would error if it were.  Comes to a R as a matrix.  (Part of that may well be RcppEigen's fault.)